DownFall/Models/DFTileSpriteNode.swift

```diff
@@ -11,31 +11,91 @@ import Foundation
+            let minecartSize = CGSize(width: width*Style.DFTileSpriteNode.Exit.minecartSizeCoefficient,
+                                      height: height*Style.DFTileSpriteNode.Exit.minecartSizeCoefficient)
+            super.init(texture: mineshaft,
+                       color: .clear,
+                       size: size)
+            
+            let trackSprite = SKSpriteNode(texture: tracks, size: size)
+            trackSprite.zPosition = Precedence.background.rawValue
+            addChild(trackSprite)
+            let minecartSprite = SKSpriteNode(texture: minecart, size: minecartSize)
+            minecartSprite.zPosition = Precedence.foreground.rawValue
+            minecartSprite.position = CGPoint.positionThis(minecartSprite.frame, inBottomOf: self.frame, padding: Style.Padding.less)
+            minecartSprite.name = "minecart"
+            addChild(minecartSprite)
+            
+        default:
+
+            super.init(texture: SKTexture(imageNamed: type.textureString()),
                    color: .clear,
                    size: CGSize.init(width: width, height: height))
+        }
     }
     
     required init?(coder aDecoder: NSCoder) { fatalError("DFTileSpriteNode init?(coder:) is not implemented") }
     
-    func indicateAboutToAttack() {
-        let blinkingSprite = SKSpriteNode(color: .yellow, size: self.size)
-        blinkingSprite.zPosition = Precedence.background.rawValue
-        let blink = SKAction.run {
-            blinkingSprite.alpha = abs(blinkingSprite.alpha - 1)
+    func removeMinecart() {
+        guard self.type == .exit else { return }
+        for child in children {
+            if child.name == "minecart" {
+                child.removeFromParent()
+            }
         }
-        let wait = SKAction.wait(forDuration: 0.2)
-        let group = SKAction.sequence([blink, wait])
-        let action = SKAction.repeatForever(group)
-        blinkingSprite.run(action)
-        self.addChild(blinkingSprite)
+    }
+    
+    func indicateSpriteWillBeAttacked() {
+        let indicatorSprite = SKSpriteNode(color: .yellow, size: self.size)
+        indicatorSprite.zPosition = Precedence.background.rawValue
+        self.addChild(indicatorSprite)
+    }
+    
+    func tutorialHighlight(){
+        if TileType.rockCases.contains(type) {
+            let whiteSprite = SKSpriteNode(color: .white, size: size)
+            whiteSprite.zPosition = Precedence.background.rawValue
+            whiteSprite.alpha = 0.75
+            self.addChild(whiteSprite)
+
+        } else {
+            let border = SKShapeNode(circleOfRadius: Style.TutorialHighlight.radius)
+            border.strokeColor = .highlightGold
+            border.lineWidth = Style.TutorialHighlight.lineWidth
+            border.position = .zero
+            border.zPosition = Precedence.menu.rawValue
+           
+            self.addChild(border)
+        }
+    }
+    
+    func showFinger() {
```

Comment:
I dont like that this was all added to the DFTileSpriteNode, i feel like it can go somewhere else.